test/Tests/src/Data/Statistics_Spec.enso

```diff
@@ -0,0 +1,132 @@
+from Standard.Base import Nothing, Vector, Number, True, Illegal_Argument_Error
+
+import Standard.Base.Data.Statistics
+from Standard.Base.Data.Statistics import all
+
+import Standard.Test
+
+# === Test Resources ===
+
+type Ord number
+
+Ord.compare_to : Ord -> Ordering
+Ord.compare_to that = this.number.compare_to that.number
+
+type No_Ord number
+
+# Tests
+
+spec =
+    number_set = [0.4, -18.56, -16.99, -16.43, -45.84, 13.44, -6.85, 9.68, -8.55, 10.87, 10.38, 33.85, -41.02, 1.87, -26.52, -13.87, -39.06, 25.92, -16.01, 42.01]
+    missing_set = number_set.map_with_index i->v->(if i % 5 == 4 then Nothing else v)
+    with_nans_set = number_set.map_with_index i->v->(if i % 5 == 4 then (if i % 10 == 9 then Number.nan else Nothing) else v)
+    text_set = ["A", "B", Nothing, "D"]
+
+    ord_set = [Ord 10, Ord 2, Nothing, Ord 9]
+    no_ord_set = [No_Ord 10, No_Ord 2, Nothing, No_Ord 9]
+    
+    double_error = 0.000001
+
+    Test.group "Statistics" <|
+        Test.specify "Should be able to count valid values" <|
+            number_set.compute . should_equal 20
+            missing_set.compute . should_equal 16
+            with_nans_set.compute . should_equal 16
+            text_set.compute . should_equal 3
+
+        Test.specify "Should be able to get minimum of maximum values" <|
+            number_set.compute Minimum . should_equal -45.84 epsilon=double_error
+            missing_set.compute Minimum . should_equal -41.02 epsilon=double_error
+            with_nans_set.compute Minimum . should_equal -41.02 epsilon=double_error
+            text_set.compute Minimum . should_equal "A"
+            number_set.compute Maximum . should_equal 42.01 epsilon=double_error
+            missing_set.compute Maximum . should_equal 33.85 epsilon=double_error
+            with_nans_set.compute Maximum . should_equal 33.85 epsilon=double_error
+            text_set.compute Maximum . should_equal "D"
+
+        Test.specify "Should be able to get sum of values" <|
+            number_set.compute Sum . should_equal -101.28 epsilon=double_error
+            missing_set.compute Sum . should_equal -81.8 epsilon=double_error
+            with_nans_set.compute Sum . should_equal -81.8 epsilon=double_error
+
+        Test.specify "Should be able to get mean of values" <|
+            number_set.compute Mean . should_equal -5.064 epsilon=double_error
+            missing_set.compute Mean . should_equal -5.1125 epsilon=double_error
+            with_nans_set.compute Mean . should_equal -5.1125 epsilon=double_error
+
+        Test.specify "Should be able to get sample variance of values" <|
+            number_set.compute Variance . should_equal 582.0137832 epsilon=double_error
+            missing_set.compute Variance . should_equal 431.0218867 epsilon=double_error
+            with_nans_set.compute Variance . should_equal 431.0218867 epsilon=double_error
```

Comment:
How were the reference values computed?

I like when tests also have just some very simple data sets (like [1,2,3]) where you can just compute the results in your head. Having more complex tests cases also sounds good, but I'm wondering where we are getting the references from.

Reply:
Excel :)

Added 1,2,3,4,5 stats too